src/main/scala/beam/agentsim/agents/planning/Tour.scala

```diff
@@ -1,28 +1,37 @@
 package beam.agentsim.agents.planning
 
-import org.matsim.api.core.v01.population.{Activity, PlanElement}
+import org.matsim.api.core.v01.population.PlanElement
 import org.matsim.utils.objectattributes.attributable.Attributes
 
-/***
-  * BEAM
-  */
-class Tour(var trips: Vector[Trip] = Vector()) extends PlanElement {
+class Tour(private var tripsInternal: Vector[Trip] = Vector()) extends PlanElement {
 
-  def originActivity(): Activity = {
-    trips.head.activity
-  }
-
-  def destActivity(): Activity = {
-    trips.reverse.head.activity
-  }
+  def trips: Seq[Trip] = tripsInternal
```

Comment:
Why was this done? Ultimately if this was done to control mutation outside of this then we may want to deep copy?

Reply:
the idea was to make the class immutable

Reply:
I guess what I am saying is that it is not immutable due to this since the items in the Seq are changable:

```
scala> class Tour(var trips: Vector[Int])
defined class Tour

scala> case class Trip(x:Int, fOpt: Option[String], parentTour: Tour)
defined class Trip

scala> val test = Seq(Trip(1,None,new Tour(Vector(1))))
test: Seq[Trip] = List(Trip(1,None,Tour@3a388769))

scala> test.head.parentTour.trips
res3: Vector[Int] = Vector(1)

scala> test.head.parentTour.trips = Vector(0)
test.head.parentTour.trips: Vector[Int] = Vector(0)

scala> test.head.parentTour.trips
res4: Vector[Int] = Vector(0)
```

Reply:
@JustinPihony I used not the most accurate word. I meant is not exposed as reassignable variable to class clients:
![Uploading image.png…]()